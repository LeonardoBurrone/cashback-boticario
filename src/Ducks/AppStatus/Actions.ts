// TODO: testar actions
import { AppStatusActionTypes, ChangeIsLoggedInAction, ChangeLoadingAction, ChangeLoadingMessageAction } from './types';

export const changeIsLoggedInAction = (payload: boolean): ChangeIsLoggedInAction => ({
  payload,
  type: AppStatusActionTypes.CHANGE_IS_LOGGED_IN
});

export const changeLoadingAction = (payload: boolean): ChangeLoadingAction => ({
  payload,
  type: AppStatusActionTypes.CHANGE_LOADING
});

export const changeLoadingMessageAction = (payload: string): ChangeLoadingMessageAction => ({
  payload,
  type: AppStatusActionTypes.CHANGE_LOADING_MESSAGE
});
