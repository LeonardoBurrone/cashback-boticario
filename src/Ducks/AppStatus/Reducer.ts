// TODO: testar reducer
import { Reducer } from 'redux';

import { AppStatusAction, AppStatusActionTypes, AppStatusState } from './types';

export const initialState: AppStatusState = {
  loading: false
};

export const appStatusReducer: Reducer<AppStatusState, AppStatusAction> = (
  state = initialState,
  action: AppStatusAction
): AppStatusState => {
  const { payload } = action;

  switch (action.type) {
    case AppStatusActionTypes.CHANGE_LOADING:
      return {
        ...state,
        loading: payload
      };

    default:
      return state;
  }
};

export default appStatusReducer;
