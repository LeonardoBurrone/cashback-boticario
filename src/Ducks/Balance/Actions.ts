import { BalanceActionTypes, ChangeBalanceAction, ChangeBalanceErrorAction, FetchBalanceAction } from './types';

export const changeBalanceAction = (payload: string): ChangeBalanceAction => ({
  payload,
  type: BalanceActionTypes.CHANGE_BALANCE
});

export const changeBalanceErrorAction = (payload: boolean): ChangeBalanceErrorAction => ({
  payload,
  type: BalanceActionTypes.CHANGE_BALANCE_ERROR
});

export const fetchBalanceAction = (): FetchBalanceAction => ({
  type: BalanceActionTypes.FETCH_BALANCE
});
