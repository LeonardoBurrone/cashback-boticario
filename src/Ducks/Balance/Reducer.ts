import { Reducer } from 'redux';

import { BalanceActionTypes, BalanceState } from './types';

export const initialState: BalanceState = {
  balance: '',
  balanceError: false
};

export const balanceReducer: Reducer<BalanceState> = (state = initialState, action): BalanceState => {
  const { payload } = action;

  switch (action.type) {
    case BalanceActionTypes.CHANGE_BALANCE:
      return {
        ...state,
        balance: payload
      };

    case BalanceActionTypes.CHANGE_BALANCE_ERROR:
      return {
        ...state,
        balanceError: payload
      };

    default:
      return state;
  }
};

export default balanceReducer;
