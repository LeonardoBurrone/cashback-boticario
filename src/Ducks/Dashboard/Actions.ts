import {
  ChangePuchasesAction,
  DashboardActionTypes,
  FetchPuschasesAction,
  FetchPuschasesErrorAction,
  Purchase
} from './types';

export const changePuchasesAction = (payload: Purchase[]): ChangePuchasesAction => ({
  payload,
  type: DashboardActionTypes.CHANGE_PURCHASES
});

export const fetchPurchasesAction = (): FetchPuschasesAction => ({
  type: DashboardActionTypes.FETCH_PURCHASES
});

export const fetchPurchasesErrorAction = (payload: boolean): FetchPuschasesErrorAction => ({
  payload,
  type: DashboardActionTypes.FETCH_PURCHASES_ERROR
});
