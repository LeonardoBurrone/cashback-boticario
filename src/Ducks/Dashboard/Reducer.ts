// TODO: testar reducer
import { Reducer } from 'redux';

import { DashboardActionTypes, DashboardState } from './types';

export const initialState: DashboardState = {
  purchases: []
};

export const dashboardReducer: Reducer<DashboardState> = (state = initialState, action): DashboardState => {
  const { payload } = action;

  switch (action.type) {
    case DashboardActionTypes.CHANGE_PURCHASES:
      return {
        ...state,
        purchases: payload
      };

    default:
      return state;
  }
};

export default dashboardReducer;
