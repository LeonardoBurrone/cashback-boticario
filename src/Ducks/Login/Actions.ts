import { AuthenticateAction, LoginAction, LoginActionTypes, LogoutAction } from './types';

export const authenticateAction = (): AuthenticateAction => ({
  type: LoginActionTypes.AUTHENTICATE
});

export const loginAction = (payload: { email: string; password: string }): LoginAction => ({
  payload,
  type: LoginActionTypes.LOGIN
});

export const logoutAction = (): LogoutAction => ({
  type: LoginActionTypes.LOGOUT
});
