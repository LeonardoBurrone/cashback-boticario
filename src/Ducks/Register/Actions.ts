import { RegisterActionTypes, RegisterPurchaseAction } from './types';

export const registerPurchaseAction = (payload: {
  code: string;
  date: string;
  price: string;
}): RegisterPurchaseAction => ({
  payload,
  type: RegisterActionTypes.REGISTER_PURCHASE
});
