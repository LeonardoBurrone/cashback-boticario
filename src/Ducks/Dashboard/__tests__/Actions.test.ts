import configureMockStore from 'redux-mock-store';
import { changePuchasesAction, fetchPurchasesAction, fetchPurchasesErrorAction } from '../Actions';
import { initialState } from '../Reducer';

describe('Test Dashboard Actions', () => {
  const mockStore = configureMockStore();
  const store = mockStore(initialState);

  beforeEach(() => {
    store.clearActions();
  });

  it('Check if "changePuchasesAction" was dispatched', () => {
    store.dispatch(
      changePuchasesAction([
        {
          cashbackPercentage: 0.01,
          cashbackValue: 1,
          date: '2021-04-01T08:15:30-03:00',
          id: 123,
          price: 100,
          status: 'Approved'
        },
        {
          cashbackPercentage: 0.01,
          cashbackValue: 1,
          date: '2021-04-01T08:11:30-03:00',
          id: 123,
          price: 50,
          status: 'Validating'
        },
        {
          cashbackPercentage: 0.01,
          cashbackValue: 1,
          date: '2021-04-01T08:19:30-03:00',
          id: 123,
          price: 30,
          status: 'Rejected'
        }
      ])
    );

    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });

  it('Check if "fetchPurchasesAction" was dispatched', () => {
    store.dispatch(fetchPurchasesAction());

    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });

  it('Check if "fetchPurchasesErrorAction" was dispatched', () => {
    store.dispatch(fetchPurchasesErrorAction(true));

    const actions = store.getActions();
    expect(actions).toMatchSnapshot();
  });
});
