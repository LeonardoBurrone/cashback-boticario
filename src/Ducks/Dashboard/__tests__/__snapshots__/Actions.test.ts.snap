// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dashboard Actions Check if "changePuchasesAction" was dispatched 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "cashbackPercentage": 0.01,
        "cashbackValue": 1,
        "date": "2021-04-01T08:15:30-03:00",
        "id": 123,
        "price": 100,
        "status": "Approved",
      },
      Object {
        "cashbackPercentage": 0.01,
        "cashbackValue": 1,
        "date": "2021-04-01T08:11:30-03:00",
        "id": 123,
        "price": 50,
        "status": "Validating",
      },
      Object {
        "cashbackPercentage": 0.01,
        "cashbackValue": 1,
        "date": "2021-04-01T08:19:30-03:00",
        "id": 123,
        "price": 30,
        "status": "Rejected",
      },
    ],
    "type": "CHANGE_PURCHASES",
  },
]
`;

exports[`Test Dashboard Actions Check if "fetchPurchasesAction" was dispatched 1`] = `
Array [
  Object {
    "type": "FETCH_PURCHASES",
  },
]
`;

exports[`Test Dashboard Actions Check if "fetchPurchasesErrorAction" was dispatched 1`] = `
Array [
  Object {
    "payload": true,
    "type": "FETCH_PURCHASES_ERROR",
  },
]
`;
